### ISSUE TITLE
Is the only way to interact with the PI3 version of OP-TEE through JTAG w/special cable?

### ISSUE THREAD

**Issue by dnovick** [OP-TEE/optee_os#1505]
I'm trying to use OP-TEE for a demo, but have run into platform issues. So, for the moment I figured I'd just do a demo on one of the stable implementations of OP-TEE. I posted a question last night about which platform is most suited for this. While waiting for a response, I went ahead and built the RPI3 version, and built the SD card as instructed.

When I boot, I'm not seeing any UI through HDMI. Nor does the device ever get onto my network through DHCP, thus allowing me to SSH into a terminal. 

Am I correct then to assume that the only way to interact with the PI is to use the JTAG mechanism described on the PI3 build instructions? I was hoping to not have to make a special cable just to do simple mods and debugging. 

I guess I assumed that some kind of non JTAG terminal support would be there by default. Was I wrong to assume this, or is there just something wrong with my SD card config (I've built it twice)?

Dave

**Comment by igoropaniuk** [comment]
Hi @dnovick,

It's possible to use both connected display via HDMI and UART (not JTAG, JTAG is used for a bit different purposes) cable.

> When I boot, I'm not seeing any UI through HDMI.

Do you see U-boot messages and Linux kernel boot log? Or there is only a black screen all the time?

> Nor does the device ever get onto my network through DHCP

To setup network via DHCP just run (despite it starts automatically during each boot, the reason why  network configuration sometimes fails is described in https://github.com/OP-TEE/optee_os/issues/1398#issuecomment-296940487)
```
/etc/init.d/udhcpc start
```
> , thus allowing me to SSH into a terminal.

But there is no any pre-configured SSH server on the standard rootfs.

Best regards,
Igor

**Comment by dnovick** [comment]
Hey there...Yeah...I meant to say UART....but was too heads down to change my posts. 

After posting this, I found the post (https://github.com/OP-TEE/optee_os/issues/1367) that describes how to change u-boot and inittab in order to make HDMI the default. I made the changes, and I'm good now. It looks like a change is pending to make this work without manual changes...but probably hasn't made it into the stable build yet.

I also saw another post regarding network connectivity....probably similar to your input. Thanks. Of course, I'll need to install an SSH server if I really end up needing it. For now, I'm just happy to have the HDMI I/O.

I'm closing this post.

Dave
